flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer.java

```diff
@@ -1182,7 +1208,7 @@ int getTransactionCoordinatorId() {
 
 	private void recycleTransactionalProducer(FlinkKafkaInternalProducer<byte[], byte[]> producer) {
 		availableTransactionalIds.add(producer.getTransactionalId());
-		producer.close();
+		producer.close(Duration.ofSeconds(0));
```

Comment:
Can we also flush the producer here? It probably does not matter at this point because this method is invoked either after committing or aborting transactions, but I just want to be safe here in case this method is called in other places in the future.

Reply:
Will do that, yes.